js/utils.js

```diff
@@ -1511,6 +1511,33 @@ Fliplet.Registry.set('dynamicListUtils', (function() {
     });
   }
 
+  function setFilterValues(options) {
+    options = options || {};
+
+    if (!options.config) {
+      return Promise.resolve();
+    }
+
+    return new Promise(function(resolve) {
+      Fliplet.User.getCachedSession()
```

Comment:
If there's no filter option with the value type `user-profile-data` then you don't need to get the user's cached session.

You should have some logic that checks through the filter options first to determine whether user session is required. This logic will end up getting expanded to check for app storage data as well when we add support for app storage filter vlaues.

Reply:
@tonytlwu 
 I used your example code https://github.com/Fliplet/fliplet-widget-dynamic-lists/pull/464#discussion_r547386465 in which `getCachedSession` will also be called every time if the case is `user-profile-data`, but I also remember that you asked that this function would not be called several times that in the end I made in this commit 
https://github.com/Fliplet/fliplet-widget-dynamic-lists/pull/464/commits/c2078e6bfa5b6716b87864548d94101e5dd79374
And what would be better to call once and use in the right case or call every time in the case

Reply:
It should only be called once, and only when it's necessary. The code would look through all filter options to find out if user profile data is needed. If so, it calls `getCachedSession` once to get the session data before looping through the filter options and applying the values. And, when applying the values, all passport entry types should be considered as you had originally done.

It seems that your several commits and reverts seem to be missing one or more of these points. We need a solution that meets all the requirements in the ticket while performing at its best.